src/util/simd/test_avx_4x64.c

```diff
@@ -0,0 +1,458 @@
+    FD_TEST( wc_test( wd_ge( x, y ), x0>=y0, x0>=y0, x1>=y1, x1>=y1, x2>=y2, x2>=y2, x3>=y3, x3>=y3 ) );
+    FD_TEST( wc_test( wd_le( x, y ), x0<=y0, x0<=y0, x1<=y1, x1<=y1, x2<=y2, x2<=y2, x3<=y3, x3<=y3 ) );
+
+    FD_TEST( wd_test( wd_czero(    c, x ), c0?0.:x0, c1?0.:x1, c2?0.:x2, c3?0.:x3 ) );
+    FD_TEST( wd_test( wd_notczero( c, x ), c0?x0:0., c1?x1:0., c2?x2:0., c3?x3:0. ) );
+
+    FD_TEST( wd_test( wd_if( c, x, y ), c0?x0:y0, c1?x1:y1, c2?x2:y2, c3?x3:y3 ) );
+
+    /* Conversion operations */
+    /* FIXME: TEST LARGE MAG CONVERSION */
+
+    FD_TEST( wc_test( wd_to_wc( x ), !(x0==0.), !(x0==0.), !(x1==0.), !(x1==0.), !(x2==0.), !(x2==0.), !(x3==0.), !(x3==0.) ) ); /* see wd_lnotnot */
+
+    FD_TEST( wf_test( wd_to_wf( x, wf( 0.f, 1.f, 2.f, 3.f, 4.f, 5.f, 6.f, 7.f ), 0 ),
+                      (float)x0, (float)x1, (float)x2, (float)x3, 4.f, 5.f, 6.f, 7.f ) );
+    FD_TEST( wf_test( wd_to_wf( x, wf( 0.f, 1.f, 2.f, 3.f, 4.f, 5.f, 6.f, 7.f ), 1 ),
+                      0.f, 1.f, 2.f, 3.f, (float)x0, (float)x1, (float)x2, (float)x3 ) );
+
+    FD_TEST( wi_test( wd_to_wi( x, wi( 0, 1, 2, 3, 4, 5, 6, 7 ), 0 ), (int)x0, (int)x1, (int)x2, (int)x3, 4, 5, 6, 7 ) );
+    FD_TEST( wi_test( wd_to_wi( x, wi( 0, 1, 2, 3, 4, 5, 6, 7 ), 1 ), 0, 1, 2, 3, (int)x0, (int)x1, (int)x2, (int)x3 ) );
+
+    FD_TEST( wi_test( wd_to_wi_fast( x, wi( 0, 1, 2, 3, 4, 5, 6, 7 ), 0 ), 
+                      (int)rint(x0), (int)rint(x1), (int)rint(x2), (int)rint(x3), 4, 5, 6, 7 ) );
+    FD_TEST( wi_test( wd_to_wi_fast( x, wi( 0, 1, 2, 3, 4, 5, 6, 7 ), 1 ),
+                      0, 1, 2, 3, (int)rint(x0), (int)rint(x1), (int)rint(x2), (int)rint(x3) ) );
+
+    FD_TEST( wu_test( wd_to_wu( x, wu(0U,1U,2U,3U,4U,5U,6U,7U), 0 ), (uint)x0,(uint)x1,(uint)x2,(uint)x3, 4U,5U,6U,7U ) );
+    FD_TEST( wu_test( wd_to_wu( x, wu(0U,1U,2U,3U,4U,5U,6U,7U), 1 ), 0U,1U,2U,3U, (uint)x0,(uint)x1,(uint)x2,(uint)x3 ) );
+
+    FD_TEST( wu_test( wd_to_wu_fast( x, wu(0U,1U,2U,3U,4U,5U,6U,7U), 0 ),
+                      (uint)rint(x0),(uint)rint(x1),(uint)rint(x2),(uint)rint(x3), 4U,5U,6U,7U ) );
+    FD_TEST( wu_test( wd_to_wu_fast( x, wu(0U,1U,2U,3U,4U,5U,6U,7U), 1 ),
+                      0U,1U,2U,3U, (uint)rint(x0),(uint)rint(x1),(uint)rint(x2),(uint)rint(x3) ) );
+
+    FD_TEST( wl_test( wd_to_wl( x ), (long)x0, (long)x1, (long)x2, (long)x3 ) );
+
+    FD_TEST( wv_test( wd_to_wv( x ), (ulong)x0, (ulong)x1, (ulong)x2, (ulong)x3 ) );
+
+    /* Reduction operations */
+
+    FD_TEST( !wc_any( wd_ne( wd_sum_all( x ), wd_bcast( x0 + x1 + x2 + x3 ) ) ) );
+    FD_TEST( !wc_any( wd_ne( wd_min_all( x ), wd_bcast( fmin( fmin( x0, x1 ), fmin( x2, x3 ) ) ) ) ) );
+    FD_TEST( !wc_any( wd_ne( wd_max_all( x ), wd_bcast( fmax( fmax( x0, x1 ), fmax( x2, x3 ) ) ) ) ) );
+
+    /* Misc operations */
+
+    wd_t m0; wd_t m1; wd_t m2; wd_t m3;
+    wd_transpose_4x4( wd_bcast( x0 ), wd_bcast( x1 ), wd_bcast( x2 ), wd_bcast( x3 ), m0, m1, m2, m3 );
```

Comment:
It might be nice to test transpose with something other than when all columns are identical